src/engine/Union.cpp

```diff
@@ -292,3 +334,185 @@ std::unique_ptr<Operation> Union::cloneImpl() const {
+    for (size_t column = 0; column < resultTable.numColumns(); column++) {
+      ColumnIndex origin = _columnOrigins.at(column).at(!left);
+      resultTable.at(resultTable.size() - 1, column) =
+          origin == NO_COLUMN ? Id::makeUndefined() : row[origin];
+    }
+  };
+  while (it1 != range1.end() && it2 != range2.end()) {
+    localVocab.mergeWith(std::span{&it1->localVocab_, 1});
+    localVocab.mergeWith(std::span{&it2->localVocab_, 1});
+    while (index1 < it1->idTable_.size() && index2 < it2->idTable_.size()) {
+      if (isSmaller<extent>(it1->idTable_.at(index1),
+                            it2->idTable_.at(index2))) {
+        pushRow(true, it1->idTable_.at(index1));
+        index1++;
+      } else {
+        pushRow(false, it2->idTable_.at(index2));
+        index2++;
+      }
+      if (requestLaziness && resultTable.size() >= chunkSize) {
+        co_yield Result::IdTableVocabPair{std::move(resultTable),
+                                          std::move(localVocab)};
+        resultTable = IdTable{getResultWidth(), allocator()};
+        resultTable.reserve(chunkSize);
+        localVocab = LocalVocab{};
+      }
+    }
+    if (index1 == it1->idTable_.size()) {
+      ++it1;
+      index1 = 0;
+    }
+    if (index2 == it2->idTable_.size()) {
+      ++it2;
+      index2 = 0;
+    }
+  }
+
+  // append the remaining elements
+  for (auto& pair :
+       processRemaining(computePermutation<true>(), it1, range1.end(),
+                        requestLaziness, index1, resultTable, localVocab)) {
+    AD_CORRECTNESS_CHECK(requestLaziness);
+    co_yield pair;
+  }
+  for (auto& pair :
+       processRemaining(computePermutation<false>(), it2, range2.end(),
+                        requestLaziness, index2, resultTable, localVocab)) {
+    AD_CORRECTNESS_CHECK(requestLaziness);
+    co_yield pair;
+  }
+  if (!requestLaziness) {
+    co_yield Result::IdTableVocabPair{std::move(resultTable),
+                                      std::move(localVocab)};
+  }
+}
+
+// _____________________________________________________________________________
+Result::Generator Union::computeResultKeepOrder(
+    bool requestLaziness, std::shared_ptr<const Result> result1,
+    std::shared_ptr<const Result> result2) const {
+  using Range = std::variant<Result::LazyResult, std::array<Wrapper, 1>>;
+  Range leftRange = result1->isFullyMaterialized()
+                        ? Range{std::array{Wrapper{result1->idTable(),
+                                                   result1->localVocab()}}}
+                        : Range{std::move(result1->idTables())};
+  Range rightRange = result2->isFullyMaterialized()
+                         ? Range{std::array{Wrapper{result2->idTable(),
+                                                    result2->localVocab()}}}
+                         : Range{std::move(result2->idTables())};
+
+  // Clang falsely reports that the this capture is unused here.
+  DISABLE_UNUSED_LAMBDA_CAPTURE_WARNINGS
```

Comment:
An alternative without these strance macros is to explicitly call
`this->computeResultKeep`....
(use `this` in a correct but redundant way).